source/client-side-encryption/client-side-encryption.rst

```diff
@@ -742,12 +766,19 @@ configuring auto encryption on a MongoClient, it is
 constructed with a MongoClient (to a MongoDB cluster containing the key
 vault collection), KMS provider configuration, keyVaultNamespace, and
 tlsOptions. It provides an API for explicitly encrypting and decrypting values,
-creating data keys, and rewrapping data keys. It does not provide an API to
-query keys from the key vault collection, as this can be done directly on the
-MongoClient.
+and managing data keys in the key vault collection.
 
 See `Why do we have a separate top level type for ClientEncryption?`_ and `Why do we need to pass a client to create a ClientEncryption?`_.
 
+When implementing behavior and error handling for key vault functions, Drivers
+SHOULD assume the presence of a unique index in the key vault collection on the
+``keyAltNames`` field with a partial index filter for only documents where
+``keyAltNames`` exists when implementing behavior of key management functions.
+Drivers MAY choose to not validate or enforce the existence of the unique index,
```

Comment:
Am I reading this correctly in that “MAY choose not to […] enforce the existence of the unique index” also means that they MAY create the unique index if it does not yet exist? Or is the idea that drivers would be allowed to error out if it doesn’t?

Reply:
"MAY choose to not validate" means drivers are not required to assert the existence of the unique index. "MAY choose to not enforce" means drivers are not required to create the unique index if it doesn't exist. Drivers are permitted to do either of these if they so desire, but given that the reason for this clause is that it is not yet clear if/how the unique index should be created/mandated by drivers (per the "Why aren't we creating a unique index in the key vault collection?" rationale), I expect not many drivers will actually do so.